src/library_pthread.js

```diff
@@ -320,17 +328,16 @@ var LibraryPThread = {
             } else if (d.cmd === 'alert') {
               alert('Thread ' + d.threadId + ': ' + d.text);
             } else if (d.cmd === 'exit') {
-              // Thread is exiting, no-op here
+              var detached = Atomics.load(HEAPU32, (d.threadId + {{{ C_STRUCTS.pthread.detached }}} ) >> 2);
+              if (detached) {
+                PThread.returnThreadToPool(worker);
```

Comment:
Perfect! Indeed in this case, if the thread had been detached, the code was not returning the worker to the pool, but only the two cases of `pthread_cancel()` and non-detached threads with `pthread_join()` were returning workers to the pool. Looks good.